### PR TITLE
R1: move button

### PR DESCRIPTION
**Pull request by pomahtri**

### CODE REVIEW

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/button.tests.js [hunk 1]
```diff
@@ -813,4 +815,99 @@ QUnit.module('Button', function() {
             });
         });
     });
+
+
+    QUnit.module('Renovaion', {
+        Button(options = {}) {
+            $('#button').dxButton(options);
+            return $('#button');
+        }
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/button.tests.js:824]: [refactoring]
This approach is not usual for us and only raise questions - it was strange to me to find `this.Button(` call in the tests. It's better to use a default approach for component initialization to keep tests consistent and readable (for old devs at least :) )

Also test naming is pure.

However, I see that u've just copied tests from `Renovation/button.tests`, so feel free to leave this refactoring to us.

File: packages/devextreme/js/__internal/ui/button/wrapper.ts [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
-import type { Button } from '../ui/button';
-import { Option } from './common/types';
-import { getImageSourceType } from '../../core/utils/icon';
-
-export default class ButtonWrapper extends Component {
+import type { dxElementWrapper } from '@js/core/renderer';
+import { getImageSourceType } from '@js/core/utils/icon';
+import ValidationEngine from '@js/ui/validation_engine';
+import { ComponentWrapper } from '@ts/core/r1/component_wrapper';
+import type { Option } from '@ts/core/r1/types';
```
Comment (ksercs) [inline packages/devextreme/js/__internal/ui/button/wrapper.ts:9]: As far as i understand your final goal is to remove R1 at all - does it mean that in future PRs u'll update these types and component_wrapper imports?

File: packages/devextreme/js/__internal/ui/button/props.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  template?: (props: { data: { icon?: string; text?: string } }) => JSX.Element;
+
+  iconTemplate?: (props) => JSX.Element;
+
+  children?: JSX.Element;
+
+  text: string;
+
+  type: 'danger' | 'default' | 'normal' | 'success';
```
Comment (ksercs) [inline packages/devextreme/js/__internal/ui/button/props.ts:29]: Shouldn't there be types from common.d.ts?

File: packages/devextreme/js/__internal/ui/button/ink_ripple.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+  constructor(props: InkRippleProps) {
+    super(props);
+    this.state = {};
+    this.hideWave = this.hideWave.bind(this);
+    this.showWave = this.showWave.bind(this);
+  }
+
+  get getConfig(): InkRippleConfig {
+    if (this.__getterCache.getConfig !== undefined) {
```
Comment (ksercs) [inline packages/devextreme/js/__internal/ui/button/ink_ripple.tsx:32]: Why do we need caching for inkRippleConfig?

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/button.tests.js (hunk 1 above)
Comment (pomahtri) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/button.tests.js:824]: > However, I see that u've just copied tests from Renovation/button.tests, so feel free to leave this refactoring to us.

Yep, leaving it to you. I just moved tests as we want to delete all Renovation/* files

File: packages/devextreme/js/__internal/ui/button/wrapper.ts (hunk 2 above)
Comment (pomahtri) [reply packages/devextreme/js/__internal/ui/button/wrapper.ts:9]: No, my goal is only to remove generation and js/renovation folder. The files you mention will be removed or refactored + renamed in future, but that's far future

File: packages/devextreme/js/__internal/ui/button/props.ts (hunk 3 above)
Comment (pomahtri) [reply packages/devextreme/js/__internal/ui/button/props.ts:29]: I moved as it was in button.tsx: https://github.com/DevExpress/DevExtreme/blob/24_2/packages/devextreme/js/renovation/ui/button.tsx#L137

File: packages/devextreme/js/__internal/ui/button/ink_ripple.tsx (hunk 4 above)
Comment (pomahtri) [reply packages/devextreme/js/__internal/ui/button/ink_ripple.tsx:32]: It's the code produced by generators, i will not change the behavior in prior to safety
All getters were cached in renovation

**Review by ksercs — APPROVED** [review]

**Pull request merged**